### ISSUE TITLE
what is the purpose of created folders in dist folder in stenciljs

### ISSUE THREAD

**Issue by ikishanoza** [ionic-team/stencil#1951]
I am not submitting any issue but just wanted to understand stencil in better way so that I can re use multiple component with effective way in my micro front end application.

Question is explained here  https://stackoverflow.com/questions/58392117/what-is-the-purpose-of-created-folders-in-dist-folder-in-stenciljs

pardon me but I am not finding nay resource online that answer my question :)

**Comment by christian-bromann** [comment]
Hey @ikishanoza 👋 

great questions! Let me answer it for you: the folders are artifacts of the output targets you have defined in your `stencil.config.ts`. Depending on which output target you use you get different folders. Let me explain each one for you.

Based on the following output targets you get the following folders in `dist`:

- using `dist-custom-elements`: you get a `/dist/types` which contains all TypeScript definitions as well as a `/dist/components` which give you a version of your Stencil components that you can import and manually register or have them being auto-registered based on your settings
- using `www`: you get a `/www` directory which is a statically generated site that renders your components based on the `html` file in `/src/index.html`
- using `dist`: this one does the most. It creates a version of your components that uses lazy loading, meaning: you can import a very lightweight script that registers all components and only when the component is being used, Stencil will load it from your server. This output target creates the following directories:
  - `cjs`: a CJS version of your lazy loaded components
  - `esm`: an ESM version of it
  - `collection`: an intermediate compile version that you can use e.g. to integrate your Stencil component into another framework. Think about it as a description of you component for other purposes but rendering on the page.
  - `<projectname>`: this is a minified version of what's in `/dist/esm`
  - `/loader`: this directory provides an entry script for loading the lightweight script for registering components, it also is responsible for applying polyfills etc.

TLDR: your certainly don't need all these folders. It really depends on how you like to use your Stencil component. We have updated our docs on publishing Stencil components, so please take a look: https://stenciljs.com/docs/publishing

Let me know if you have further questions!

Cheers
